src/table.ts

```diff
@@ -778,133 +749,62 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
       rowStream?.removeListener('end', originalEnd);
     };
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    userStream.end = (chunk?: any, encoding?: any, cb?: () => void) => {
-      rowStreamUnpipe(rowStream, userStream);
-      userCanceled = true;
-      if (activeRequestStream) {
-        activeRequestStream.abort();
-      }
-      if (retryTimer) {
-        clearTimeout(retryTimer);
-      }
-      return originalEnd(chunk, encoding, cb);
-    };
-
-    const makeNewRequest = () => {
-      // Avoid cancelling an expired timer if user
-      // cancelled the stream in the middle of a retry
-      retryTimer = null;
-
-      const lastRowKey = chunkTransformer ? chunkTransformer.lastRowKey : '';
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      chunkTransformer = new ChunkTransformer({decode: options.decode} as any);
-
-      const reqOpts = {
-        tableName: this.name,
-        appProfileId: this.bigtable.appProfileId,
-      } as google.bigtable.v2.IReadRowsRequest;
-
-      const retryOpts = {
-        currentRetryAttempt: 0, // was numConsecutiveErrors
-        // Handling retries in this client. Specify the retry options to
-        // make sure nothing is retried in retry-request.
-        noResponseRetries: 0,
-        shouldRetryFn: (_: any) => {
-          return false;
-        },
-      };
-
-      if (lastRowKey) {
-        // Readjust and/or remove ranges based on previous valid row reads.
-        // Iterate backward since items may need to be removed.
-        for (let index = ranges.length - 1; index >= 0; index--) {
-          const range = ranges[index];
-          const startValue = is.object(range.start)
-            ? (range.start as BoundData).value
-            : range.start;
-          const endValue = is.object(range.end)
-            ? (range.end as BoundData).value
-            : range.end;
-          const startKeyIsRead =
-            !startValue ||
-            TableUtils.lessThanOrEqualTo(
-              startValue as string,
-              lastRowKey as string
-            );
-          const endKeyIsNotRead =
-            !endValue ||
-            (endValue as Buffer).length === 0 ||
-            TableUtils.lessThan(lastRowKey as string, endValue as string);
-          if (startKeyIsRead) {
-            if (endKeyIsNotRead) {
-              // EndKey is not read, reset the range to start from lastRowKey open
-              range.start = {
-                value: lastRowKey,
-                inclusive: false,
-              };
-            } else {
-              // EndKey is read, remove this range
-              ranges.splice(index, 1);
-            }
-          }
+    (() => {
```

Comment:
Could you add some more comments to this block?

Reply:
Done.